### DISCUSSION TITLE
Provide as well as sink

### DISCUSSION THREAD

**Discussion by PeterWone** [microsoft/vscode-discussions#996] [Extension Development QnA] [answered]
There are plenty of ways for an extension to provide information _to_ vscode but there generally isn't a corresponding way for an extension to consume the same information. Some examples:

- People want my printing extension to use the same colour scheme as they set for the editor. There's no way to request the CSS that applies to either active editor or a specific langId.
- People keep asking me to respect folding state while printing. I tell them that this isn't part of the document, it's editor state and vscode won't give it to me. Perhaps I'm wrong about that but I can't find any pertinent docs.

This isn't specifically about these two examples (editors can do dark mode but paper is always white, so using editor CSS won't always work). It's about the general principal that extensions should not know about each other and therefore for any context in which vscode is a sink, it should also be a provider. If _you_ need the info then eventually an extension will want it.

**Comment by isidorn** [comment, marked as answer]
This is a fair feature request, but not something we have on the plan right now to add. Thanks for understanding.
